great_expectations/compatibility/aws.py

```diff
@@ -0,0 +1,29 @@
+from __future__ import annotations
+
+from great_expectations.compatibility.not_imported import NotImported
+
+AWS_NOT_IMPORTED = NotImported(
+    "Unable to load AWS connection object; please 'pip install boto3 botocore'"
+)
+
+try:
+    import boto3
+except ImportError:
+    boto3 = AWS_NOT_IMPORTED
+
+try:
+    import botocore
+except ImportError:
+    botocore = AWS_NOT_IMPORTED
+
+try:
```

Comment:
@Shinnnyshinshin I would be really delighted to see if we could succeed in changing this pattern to be as importing exceptions from botocore -- as `exceptions` -- and then we can access any exception (including `ClientError`).  We may have working examples in `Google` and `Spark` compatibility modules (and `SQLAlchemy` as well).  Thank you!

Reply:
oo great idea. I'll update the PR and ping you

Reply:
Actually not seeing this one @alexsherstinsky could you point me to the pattern you're thinking of?